providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -25,7 +27,40 @@ export class SendgridEmailProvider implements IEmailProvider {
   async sendMessage(
     options: IEmailOptions
   ): Promise<ISendMessageSuccessResponse> {
-    const response = await this.sendgridMail.send({
+    const mailData = this.createMailData(options);
+    const response = await this.sendgridMail.send(mailData);
+
+    return {
+      id: response[0]?.headers['x-message-id'],
+      date: response[0]?.headers?.date,
+    };
+  }
+
+  async checkIntegration(
+    options: IEmailOptions
+  ): Promise<ICheckIntegrationResponse> {
+    try {
+      const mailData = this.createMailData(options);
+
+      const response = await this.sendgridMail.send(mailData);
+
+      if (response[0]?.statusCode === 202) {
+        return {
+          success: true,
+          message: 'Integration Successful',
+        };
+      }
```

Comment:
It looks like in case the message is 200, we will return a void message. Could we just remove this `if` statement and rely on the sendgrid Throw mechanism?